meson.build

```diff
@@ -742,7 +742,7 @@ endif
 time_epoch = get_option('time-epoch')
 if time_epoch == -1
         time_epoch = run_command(sh, '-c', 'echo "$SOURCE_DATE_EPOCH"').stdout().strip()
-        if time_epoch == '' and git.found() and run_command('test', '-e', '.git').returncode() == 0
+        if time_epoch == '' and git.found() and fs.is_dir('.git')
```

Comment:
I think this would break git workspaces, as in that case it's a file, not a directory. The previous check only checked for existence.